src/algorithms/timestep/clusterexpansion.jl

```diff
@@ -0,0 +1,164 @@
+struct ClusterExpansion
+    N::Int
+end
+
+function make_time_mpo(H::MPOHamiltonian, dt::Number, alg::ClusterExpansion)
+    N = alg.N
+    lmax = N ÷ 2 # largest level
+    τ = -im * dt
+    # spaces
+    P = physicalspace(H)[1]
+    D = dim(physicalspace(H)[1]) # physical dimension
+    V = BlockTensorKit.oplus([ℂ^(D^2l) for l in 0:lmax]...)
+
+    TT = tensormaptype(ComplexSpace, 2, 2, ComplexF64)
+    O = SparseBlockTensorMap{TT}(undef, V ⊗ P ← P ⊗ V)
+
+    for n in 1:N
+        if n == 1
+            O[1, 1, 1, 1] = _make_Hamterms(H, 1, τ)
+        elseif n == 2
+            U, S, V = tsvd(_make_b(H, O, 2, τ), ((1, 2, 4), (3, 5, 6)))
+            O[1, 1, 1, 2] = permute(U * sqrt(S), ((1, 2), (3, 4)))
+            O[2, 1, 1, 1] = permute(sqrt(S) * V, ((1, 2), (3, 4)))
+        else
+            nl = n ÷ 2 # new level
+            jnl = nl + 1 # Julia indexing
+            apply_function = _make_apply_functions(O, n)
+            b = _make_b(H, O, n, τ)
+            Onew, info = lssolve(apply_function, b; verbosity=5)
+
+            (info.converged) != 1 &&
+                @warn "Did not converge when constucting the $n cluster"
+
+            if isodd(n) # linear problem
+                # assign the new level to O
+                O[jnl, 1, 1, jnl] = Onew
+            elseif iseven(n) # linear problem + svd
+                U, S, V = tsvd(Onew, (1, 2, 4), (3, 5, 6))
+                # assign the new levels to O
+                O[jnl - 1, 1, 1, jnl] = permute(U * sqrt(S), ((1, 2), (3, 4)))
+                O[jnl, 1, 1, jnl - 1] = permute(sqrt(S) * V, ((1, 2), (3, 4)))
+            end
+        end
+    end
+
+    # return O
+    return MPO(PeriodicVector([O]))
+end
+
+function _make_Hamterms(H::MPOHamiltonian, n::Int, τ::Number)
+    return add_util_leg(convert(TensorMap,
+                                scale(DenseMPO(open_boundary_conditions(H, n)), τ)))
+end
+
+function _make_b(H::MPOHamiltonian, O::SparseBlockTensorMap, n::Int, τ::Number)
+    # be - bo
+    be = permute(exp(permute(_make_Hamterms(H, n, τ), Tuple(1:(n + 1)),
+                             Tuple(vcat([2n + 2], collect((n + 2):(2n + 1)))))),
+                 Tuple(1:(n + 1)), Tuple(vcat(collect((n + 3):(2n + 2)), [n + 2])))
+    bo = _fully_contract_O(O, n)
+    return be - bo
+end
+
+function _fully_contract_O(O::SparseBlockTensorMap, n::Int)
+    # Make projector onto the 0 subspace of the virtual level
+    Pl = zeros(ComplexF64, ℂ^1 ← space(O, 1))
+    Pl[1] = 1
```

Comment:
This pattern seems wrong (and probably will break once I actually get to fixing that), and probably only works for non-symmetric tensors... If I get this right, you want a trivial unit tensor as the first element, so you should probably just instantiate that tensor. There are some examples of this scattered throughout MPSKit 😉